contrib/ci/check-dep-versions.py

```
@@ -0,0 +1,303 @@
+    details = "; ".join(errors) if errors else "no usable build directory found"
+    raise RuntimeError(
+        "Unable to run `meson introspect --dependencies`. "
+        "Pass --builddir or set MESON_BUILD_DIR. "
+        f"Details: {details}"
+    )
+
+
+def get_xml_versions(xml_file: str) -> Dict[str, str]:
+    versions: Dict[str, str] = {}
+
+    tree = etree.parse(xml_file)
```

Ruff (0.15.10)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)